ports/rp2/modrp2.c

```diff
@@ -50,11 +55,50 @@ STATIC mp_obj_t rp2_country(size_t n_args, const mp_obj_t *args) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(rp2_country_obj, 0, 1, rp2_country);
 #endif
 
+// Improved version of
+// https://github.com/raspberrypi/pico-examples/blob/master/picoboard/button/button.c
+STATIC bool __no_inline_not_in_flash_func(bootsel_button)() {
+    const uint CS_PIN_INDEX = 1;
+
+    // Disable interrupts and the other core since they might be
+    // executing code from flash and we are about to temporarily
+    // disable flash access.
+    mp_uint_t atomic_state = MICROPY_BEGIN_ATOMIC_SECTION();
+
+    // Set the CS pin to high impedance.
+    hw_write_masked(&ioqspi_hw->io[CS_PIN_INDEX].ctrl,
+        (GPIO_OVERRIDE_LOW << IO_QSPI_GPIO_QSPI_SS_CTRL_OEOVER_LSB),
+        IO_QSPI_GPIO_QSPI_SS_CTRL_OEOVER_BITS);
+
+    // Delay without calling any functions in flash.
+    uint32_t start = timer_hw->timerawl;
+    while ((uint32_t)(timer_hw->timerawl - start) <= MICROPY_HW_BOOTSEL_DELAY_US) {
```

Comment:
Would it be worth making this delay configurable via a parameter passed in to this function?  Then Python code could do, eg `rp2.bootsel_button(4)` to select 4us.

Reply:
I don't think it will be very common for people to be attaching extra hardware to the BOOTSEL pin that affects the timing, so the appropriate timing is probably just a function of the board they are using and what capacitors and resistors it has.  So this pull request lets you set the delay in a board configuration file like `ports/rp2/boards/PICO/mpconfigboard.h` and I think that is probably good enough.

I don't anticipate many users needing to increase the delay, and those who want to decrease it to save a few microseconds probably would want to avoid whatever overhead there is for passing a parameter.

Reply:
OK, that makes sense.

We can always add an argument in the future if it becomes necessary.